common/voxel/chunkmanager.cpp

```diff
@@ -236,18 +362,81 @@ Vector3* ChunkManager::translateIndexToPosition(
             2); // Calculating the Z position of the Chunk in the World
   return new Vector3(ara[0], ara[1], ara[2]);
 }
+*/
+
+std::string ChunkManager::generateKey(int sx, int sy, int sz)//translates a chunk position value, to a standardized hash map key
+{
+	std::stringstream ssKey;				//creating a stringstream object to translate the integers to a string
+	ssKey << sx << "," << sy << "," << sz;	//standard format for the hash map keys is 'x,y,z'
+	return ssKey.str();						//returning the formatted string
+}
+
+Vector3 ChunkManager::keyToVector(std::string key)//translates a hash table key to a Vector3 for reduce operations
+{
+	std::vector<int> result;
+	std::stringstream data(key);
+	std::string line;
+	while(std::getline(data,line,','))
+	{
+		result.push_back(std::stoi(line));
+	}
+	float x_pos = result[0];
+	float y_pos = result[1];
+	float z_pos = result[2];
+	
+	Vector3 chunk(x_pos, y_pos, z_pos);
+	return chunk;
+}
+
```

Comment:
>The client does not need to know how we track everything.

Let's see if I understand this correctly. If the caller wanted to know which block is at position (1, 3, 4) in the world, they would have to somehow divine which public `ChunkList` to work with--let's say they choose `chunkProcessingList`, because why not--and then call something like:
```c++
// Get chunk at position (0, 0, 0)
Chunk* chunk = chunkManager->chunkProcessingList[chunkManager->generateKey(0, 0, 0)];
```
The caller *had* to know about that `generateKey` nonsense--which is confusing--but now at least they have their 16x16x16 chunk of voxels/blocks to work with. This is nice and all, but then they'd have no way of interacting with the individual blocks, because `Chunk` has no public methods that return what block is at some specified (x, y, z) location. 

But let's suppose we added a member function to do this called `getBlock`. They could then call:
```c++
Block block = chunk->getBlock(1, 3, 4);
```
And now they can work with the block at position (1, 3, 4) in the world.

The thing is, I don't see how this is simpler than the ChunkManager's already implemented function `get`.
```c++
auto block = chunkManager->get(1, 3, 4); // returns block at position (1, 3, 4).
```
Which avoids the caller having to know anything about chunks under the hood.

>The use of the string is a consequence of using a concurrent_hash_map.

I don't see why you'd use a hash map to look up Chunks. I recall someone in class claiming it's "faster" than the current vector implementation without any evidence. Honestly, I'd expect it to be slower, because you'd have to:

1. Create a std::string from an (x, y, z) coordinate.
2. Hash the string and perform a string comparison with the key-value pair already in the map.

It's an unnecessary level of indirection when you can just convert a given (x, y, z) to it's appropriate location in a vector.